src/libsyntax/diagnostic.rs

```diff
@@ -293,8 +293,8 @@ impl fmt::Display for Level {
 
         match *self {
             Bug => "error: internal compiler error".fmt(f),
-            Fatal | Error => "error".fmt(f),
-            Warning => "warning".fmt(f),
+            Fatal | Error => "\n  error".fmt(f),
+            Warning => "\n  warning".fmt(f),
```

Comment:
One slight issue with full paths is that when cargo spits out errors for multiple deps it's confusing which dep is being talked about.